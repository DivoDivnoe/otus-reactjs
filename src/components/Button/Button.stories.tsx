import React from 'react';
import { ComponentStory, ComponentMeta } from '@storybook/react';

import { Button } from './Button';

export default {
  title: 'Example/Button',
  component: Button,
  args: {
    children: 'Button',
    isDisabled: false,
  },
  argTypes: {
    clickHandler: { action: 'clicked' },
  },
} as ComponentMeta<typeof Button>;

const Template: ComponentStory<typeof Button> = (args) => <Button {...args} />;

export const Inactive = Template.bind({});
Inactive.args = { isActive: false };

export const Active = Template.bind({});
Active.args = { isActive: true };
