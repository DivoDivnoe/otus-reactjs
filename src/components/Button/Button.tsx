import React, { FC } from 'react';
import styled from '@emotion/styled';

interface ButtonProps {
  children?: string;
  isActive: boolean;
  isDisabled?: boolean;
  clickHandler: () => void;
}

const ButtonItem = styled.button`
  box-sizing: border-box;
  padding-top: 2px;
  padding-bottom: 2px;
  padding-left: 6px;
  padding-right: 6px;
  font-size: 16px;
  line-height: 22px;
  font-family: Arial, Helvetica, sans-serif;
  color: #ffffff;
  text-align: center;
  border: none;
  border-radius: 7px;
  cursor: pointer;
  background-color: ${(props: Omit<ButtonProps, 'clickHandler'>) =>
    props.isActive ? 'lightblue' : 'blue'};
  pointer-events: ${(props: Omit<ButtonProps, 'clickHandler'>) =>
    props.isDisabled ? 'none' : 'auto'};
`;

const Button: FC<ButtonProps> = ({
  isActive,
  isDisabled,
  clickHandler,
  children,
}) => {
  return (
    <ButtonItem
      isDisabled={isDisabled}
      isActive={isActive}
      onClick={clickHandler}
    >
      {children}
    </ButtonItem>
  );
};

export default Button;
