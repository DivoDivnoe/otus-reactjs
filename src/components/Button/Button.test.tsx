import React from 'react';
import { render, screen, fireEvent } from '@testing-library/react';
import Button from './Button';

describe('Button', () => {
  it('renders Button component with correct text', () => {
    const mocks = {
      isActive: false,
      clickHandler: jest.fn(),
    };
    const text = 'Run';
    render(<Button {...mocks}>{text}</Button>);

    expect(screen.getByText('Run')).toBeInTheDocument();
  });

  it('handles click event correctly', () => {
    const mocks = {
      isActive: false,
      clickHandler: jest.fn(),
    };
    const text = 'Run';
    render(<Button {...mocks}>{text}</Button>);

    fireEvent.click(screen.getByText('Run'));
    expect(mocks.clickHandler).toHaveBeenCalledTimes(1);
  });
});
