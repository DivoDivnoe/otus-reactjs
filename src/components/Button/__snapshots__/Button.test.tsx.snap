// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component renders correctly 1`] = `
<DocumentFragment>
  <button
    class="css-som8ht"
  >
    Run
  </button>
</DocumentFragment>
`;
